### PR TITLE
TH-935 | Fix double 302 redirection

### PR DESCRIPTION
**Pull request by Sartonon**
## Description :sparkles:

- redirect "/" straight to "/:locale/home"

## Screenshots :camera_flash:

Screenshot 2021-03-02 at 19 57 24

## Additional notes :spiral_notepad:

### CODE REVIEW

**Review by nikomakela — COMMENTED** [review]
Looks good to me, but I'm not very familiar with Router yet.

File: src/constants.ts [hunk 1]
```diff
@@ -40,6 +40,12 @@ export enum SUPPORT_LANGUAGES {
   EN = 'en',
 }
 
+export const supportedLanguages = [
```
Comment (nenonja) [inline src/constants.ts:43]: `export const supportedLanguages = Object.values(SUPPORT_LANGUAGES);` is shorter and safer too if SUPPORT_LANGUAGES are updated?

Comment (nenonja) [reply src/constants.ts:43]: Also, there is dozens of `Object.values(SUPPORT_LANGUAGES)` in the codebase, they could be replaced with `supportedLanguages`

**Review by nenonja — COMMENTED** [review]
I'm not guru with redirects at all but start reading about redirects:
https://www.drlinkcheck.com/blog/http-redirects-301-302-303-307-308#:~:text=Use%20a%20302%20redirect%20when,know%20what%20you%20are%20doing.
and it strongly suggest _not_ to use 308 code:
`Due to still limited support of 308, the recommendation is to always go with 301 redirects, unless you require POST requests to be redirected properly and are certain that all clients understand the 308 response code.`

Also I dont understand why `/fi -> /fi/home` is permanent redirect but `/` -> `/fi/home` is temporary redirect?

File: src/constants.ts (hunk 1 above)
Comment (Sartonon) [reply src/constants.ts:43]: Yeah should use object.values

**Comment by nenonja** [comment]
> `Due to still limited support of 308, the recommendation is to always go with 301 redirects, unless you require POST requests to be redirected properly and are certain that all clients understand the 308 response code.`

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/308
It seems that only IE 11 with windows 7 or 8 or older does not work. Perhaps it's good enough for us.

**Comment by nenonja** [comment]
Conflict must be resolved and hopefully browsert tests turn to green after rebasing, master has fixes to them.

**Comment by Sartonon** [comment]
> I'm not guru with redirects at all but start reading about redirects:
> 
> https://www.drlinkcheck.com/blog/http-redirects-301-302-303-307-308#:~:text=Use%20a%20302%20redirect%20when,know%20what%20you%20are%20doing.
> 
> and it strongly suggest _not_ to use 308 code:
> 
> `Due to still limited support of 308, the recommendation is to always go with 301 redirects, unless you require POST requests to be redirected properly and are certain that all clients understand the 308 response code.`
> 
> 
> 
> Also I dont understand why `/fi -> /fi/home` is permanent redirect but `/` -> `/fi/home` is temporary redirect?
> 
> 

Me neither. That is what was told in SEO report.. maybe we need to ask them again

**Comment by Sartonon** [comment]
Screenshot 2021-03-05 at 14 30 10

File: src/server/__tests__/utils.test.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { handleReactRouterRedirection } from '../utils';
+
+describe('handleReactRouterRedirection', () => {
+  it('redirects to /fi/home 308', () => {
```
Comment (nenonja) [inline src/server/__tests__/utils.test.tsx:5]: very small thing but description could be `"redirects '/fi' to '/fi/home' (308 Permanent Redirect)"` and 
`"redirects '/' to '/fi/home' (302 Temporary Redirect)"`

**Review by nenonja — APPROVED** [review]
Looks good! One small improvement suggestion, but not mandatory

File: src/server/__tests__/utils.test.tsx (hunk 2 above)
Comment (Sartonon) [reply src/server/__tests__/utils.test.tsx]: fixed :)

**Pull request merged**